### PR TITLE
Amputee and Bad Teeth

### PR DESCRIPTION
**Pull request by Xenu47**
TL;DR: Amputee no longer applies burns and zombies no longer can infect players by biting the left arm. Bad Teeth now properly adds pain, without overwriting existing.

## Problem with Amputee was: It didn't remove burns

Function for just removing burns was `bodyPart:setBurnTime(0)`
But while testing in debug and applied Scratch via health panel with cheat mode, error popped out. `setScratched()` was not working idk why, STACK TRACE was not helpful to see what went wrong:
```
---
STACK TRACE
---
Callframe at: setScratched
*(location of setScratched())*
```

### What commits include
Found another function `bodyPart:RestoreToFullHealth()`, which does everything amputee needs, except removing dirt and blood.
Therefore I also added code to _clean_ left arm (`player:getVisual()...`, `player:resetModelNextFrame()`), because why not and who knows, maybe one day you will remove left arm model, but blood and dirt will still apply.

Additionally, zombie bites provide Infection to Player, not to bodyPart (which is already being removed with `bodyPart:RestoreToFullHealth()`). 
To remove infection from Player, I added extra check if player **got the infection with recent damage (!)**. If yes and to the left arm - remove infection from player.

## Problem with Bad Teeth was: It was overwriting pain if player had it from other source

First of all, in-game General Debuggers state: 
image
And indeed, I could not move pain slider.
Secondly, i browsed actual game code, every time Zomboid tries to increase your pain it does so on bodyPart via `bodyPart:setAdditionalPain(float pain)` or `self.otherPlayer:getOnlineID(), self.bodyPart:getIndex(), addPain)`

### What commits include
A check to set pain only once per food consumption, by comparing previous and current `healthtimer`'s

> before, it just added ~1000/90=10 pain every in-game tick,  but function for adding pain was wrong and it just did some funny stuff: as from I saw, it just added a pop-up, but not actually increased pain level o_O

Now it works by adding 'additional pain' to the Head, without any overwrites:
`Head:setAdditionalPain(Head:getAdditionalPain() + pain);`

### CODE REVIEW

**Comment by Xenu47** [comment]
Superimmune part:

Player isInfected, but wound have no indication of it (no orange text): 
image

Roll is 100%, but can not fully cure:
image

Tested by healing wound to full - infection disappeared. 

I assume bite inflicts some addtional infection that is strictly bound to wound. 
Since Superimmune have to clear all infections from wounds:
> Your wounds can never become infected

I added extra line to clear wound infection: `b:SetInfected(false);`

File: Contents/mods/More Traits/media/lua/client/MoreTraits.lua [hunk 1]
```diff
@@ -871,11 +873,21 @@ local function indefatigablecounter()
...
-        if player:getBodyDamage():getHealthFromFoodTimer() > 1000 then
-            player:getStats():setPain(player:getBodyDamage():getHealthFromFoodTimer() / 90);
+        if healthtimer > 1000 then
+            if healthtimer > playerdata.fPreviousHealthFromFoodTimer then
+                local Head = player:getBodyDamage():getBodyPart(BodyPartType.FromString("Head"));
+                local pain = player:getBodyDamage():getHealthFromFoodTimer() * 0.01;
+                Head:setAdditionalPain(Head:getAdditionalPain() + pain);
+                print("pain: " .. pain)
+                print("healthtimer: " .. healthtimer);
+                print("previous ht: " .. playerdata.fPreviousHealthFromFoodTimer);
```
Comment (hypnotoadtrance) [inline Contents/mods/More Traits/media/lua/client/MoreTraits.lua:888]: I would suggest removing these print commands to avoid log spam.

File: Contents/mods/More Traits/media/lua/client/MoreTraits.lua [hunk 2]
```diff
@@ -1270,28 +1282,31 @@ local function amputee(_player)
...
-            ForeArm_L:setBleeding(false);
-            ForeArm_L:setHaveGlass(false);
-            ForeArm_L:SetInfected(false);
+            ForeArm_L:RestoreToFullHealth();
+            if justGotInfected then
+                player:getBodyDamage():setInfected(false);
+            end
+            player:getVisual():setBlood(BloodBodyPartType.FromIndex(BodyPartType.ToIndex(ForeArm_L:getType())), 0);
+            player:getVisual():setDirt(BloodBodyPartType.FromIndex(BodyPartType.ToIndex(ForeArm_L:getType())), 0);
+            player:resetModelNextFrame();
```
Comment (hypnotoadtrance) [inline Contents/mods/More Traits/media/lua/client/MoreTraits.lua:1300]: Will resetting the model cause the player to reset all animation states? It might cause unforseen issues.

Comment (Xenu47) [reply Contents/mods/More Traits/media/lua/client/MoreTraits.lua:1300]: Just checked with exercises and building, at least with these activities this code does interrupt animations. 
If you fear unforseen, this is not the most important feature to be keeped.

File: Contents/mods/More Traits/media/lua/client/MoreTraits.lua (hunk 1 above)
Comment (Xenu47) [reply Contents/mods/More Traits/media/lua/client/MoreTraits.lua:888]: These print commands trigger once per consumed food, if this food also causes players teeth to hurt.

**Comment by hypnotoadtrance** [comment]
Very good!
